### PR TITLE
Fixes for coverity warnings

### PR DESCRIPTION
**Pull request by mwleeds**

### CODE REVIEW

File: common/flatpak-dir.c [hunk 1]
```diff
@@ -8662,19 +8668,20 @@ flatpak_dir_deploy_update (FlatpakDir        *self,
...
   old_subpaths = flatpak_deploy_data_get_subpaths (old_deploy_data);
 
-  previous_ids = g_strdupv ((char **) flatpak_deploy_data_get_previous_ids (old_deploy_data, NULL));
+  previous_ids = flatpak_deploy_data_get_previous_ids (old_deploy_data, NULL);
+  previous_ids_owned = g_strdupv ((char **) previous_ids);
   if (opt_previous_ids)
     {
-      g_auto(GStrv) old_previous_ids = previous_ids;
-      previous_ids = flatpak_strv_merge (old_previous_ids, (char **) opt_previous_ids);
+      previous_ids_owned = flatpak_strv_merge ((char **) previous_ids, (char **) opt_previous_ids);
```
Comment (alexlarsson) [inline common/flatpak-dir.c:8668]: This seems to overwrite the above value of previous_ids_owned, thus leaking.

File: common/flatpak-dir.c [hunk 2]
```diff
@@ -8662,19 +8668,20 @@ flatpak_dir_deploy_update (FlatpakDir        *self,
...
+      g_clear_pointer (&previous_ids, g_free);
```
Comment (alexlarsson) [inline common/flatpak-dir.c:8669]: Why is this needed whe previous_ids is already a autofree?

File: common/flatpak-dir.c [hunk 3]
```diff
@@ -12465,12 +12466,11 @@ find_matching_ref (GHashTable  *refs,
             g_string_append (err, ", ");
 
           const char *branch = flatpak_decomposed_get_branch (ref);
-
-          g_string_append (err,
```
Comment (alexlarsson) [inline common/flatpak-dir.c:12462]: Not a huge issue, but this could just g_string_append_printf.

File: common/flatpak-progress.c [hunk 4]
```diff
@@ -192,6 +192,7 @@ update_status_progress_and_estimating (FlatpakProgress *self)
      extra data, so ignore those */
   if (self->requested == 0)
     {
+      g_string_free (buf, TRUE);
```
Comment (alexlarsson) [inline common/flatpak-progress.c:195]: Alternatively, move the allocation to after the check.

File: app/flatpak-builtins-document-export.c [hunk 5]
```diff
@@ -174,6 +174,7 @@ flatpak_builtin_document_export (int argc, char **argv,
   fd_list = g_unix_fd_list_new ();
   fd_id = g_unix_fd_list_append (fd_list, fd, error);
   close (fd);
+  fd = -1;
```
Comment (alexlarsson) [inline app/flatpak-builtins-document-export.c:177]: I would prefer close(glnx_steal_fd (&fd))

**Comment by alexlarsson** [comment]
Other than the comments it looks good to me.

File: common/flatpak-dir.c (hunk 2 above)
Comment (mwleeds) [reply common/flatpak-dir.c:8669]: It's not strictly needed but it seems more robust this way so that someone can't inadvertently add code to use the now-outdated value.

File: common/flatpak-progress.c (hunk 4 above)
Comment (mwleeds) [reply common/flatpak-progress.c:195]: Yeah that seems cleaner

File: app/flatpak-builtins-document-export.c (hunk 5 above)
Comment (mwleeds) [reply app/flatpak-builtins-document-export.c:177]: The docs for `glnx_close_fd()` say it is effectively `close(glnx_steal_fd (&fd))` but also with a check for `EBADF` so I'll go with that.

**Comment by mwleeds** [comment]
Pushed a fix for the flatpak_deploy_data_get_previous_ids() docs

**Review by sethash20 — APPROVED** [review]
